### ISSUE TITLE
False positive on sorted_first_last with Realm

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#2533]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ echo "results.sorted(byKeyPath: #keyPath(Message.timestamp), ascending: false).first" | swiftlint lint --no-cache --use-stdin --enable-all-rules
Loading configuration from '.swiftlint.yml'
<nopath>:1:1: warning: Min or Max over Sorted First or Last Violation: Prefer using `min()` or `max()` over `sorted().first` or `sorted().last` (sorted_first_last)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.29.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew
* Paste your configuration file:

```yml
opt_in_rules:
  - sorted_first_last
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.

No

* Which Xcode version are you using (check `xcode-select -p`)?

10.2

* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This triggers a violation:
results.sorted(byKeyPath: #keyPath(Message.timestamp), ascending: false).first
```
